service/src/main/java/bio/terra/workspace/service/workspace/flight/StoreGcpContextStep.java

```diff
@@ -27,8 +29,18 @@ public StoreGcpContextStep(GcpCloudContextService gcpCloudContextService, UUID w
   public StepResult doStep(FlightContext flightContext) throws InterruptedException {
     String projectId = flightContext.getWorkingMap().get(GCP_PROJECT_ID, String.class);
 
-    // Create the cloud context; throws if the context already exists. We let
-    // Stairway handle that.
+    // Create the cloud context; throws if another flight already created the context.
```

Comment:
No change requested, just an observation. I think you could also do this (check for flight id, write new context) in a single db call using a constraint on the table that includes the flight id and then modifying the create context SQL statement to do nothing, instead of throwing an exception, if that constraint is violated. That might be useful to look into if we ever run into db contention issues.

Reply:
It is a good idea. The column is nullable, but Postgres supports nulls - each considered unique from the other - in a unique constraint. So it could be done.